pkg/comp-functions/functions/miniobucket/miniobucket.go

```diff
@@ -21,54 +20,47 @@ const (
 // ProvisionMiniobucket will create a bucket in a pre-deployed minio instance.
 // This function will leverage provider-minio to deploy proper policies and users
 // alongside the bucket.
-func ProvisionMiniobucket(ctx context.Context, iof *runtime.Runtime) runtime.Result {
+func ProvisionMiniobucket(ctx context.Context, svc *runtime.ServiceRuntime) *xfnproto.Result {
 
 	bucket := &appcatv1.ObjectBucket{}
 
-	err := iof.Desired.GetComposite(ctx, bucket)
+	err := svc.GetObservedComposite(bucket)
 	if err != nil {
-		return runtime.NewFatalErr(ctx, "Could not get ObjectBucket claim", err)
+		return runtime.NewFatalResult(err)
 	}
 
-	config, ok := iof.Config.Data["providerConfig"]
+	config, ok := svc.Config.Data["providerConfig"]
 	if !ok {
-		return runtime.NewFatalErr(ctx, "no providerConfig specified in composition config", fmt.Errorf("no providerConfig specified"))
```

Comment:
Do I understand correctly, the error description is not relevant anymore?

Reply:
Yes, the new API just takes an error and no description.

If you want to add additional information to the error you can do it with standard golang error nesting.

Also, if you just want a default result you can just return nil, it handles a lot closer how vanilla error handling works.